numpy/core/src/umath/matmul.c.src

```diff
@@ -494,9 +499,56 @@ NPY_NO_EXPORT void
                  * non-blasable (or non-ccontiguous output)
                  * we could still use BLAS, see gh-12365.
                  */
-                @TYPE@_matmul_inner_noblas(ip1, is1_m, is1_n, 
-                                           ip2, is2_n, is2_p,
-                                           op, os_m, os_p, dm, dn, dp);
+                if(tmp_ip1 == NULL) {
+                    tmp_ip1 = (@typ@ *)malloc(sizeof(@typ@) * dn * dm);
+                }
+                if(tmp_ip2 == NULL) {
+                    tmp_ip2 = (@typ@ *)malloc(sizeof(@typ@) * dn * dp);
+                }
+                if(tmp_op == NULL) {
+                    tmp_op = (@typ@ *)malloc(sizeof(@typ@) * dm * dp);
+                }
+
+                /* Not enough memory */
+                if(
+                    tmp_ip1 == NULL || tmp_ip2 == NULL || tmp_op == NULL
+                ) {
+                    @TYPE@_matmul_inner_noblas(ip1, is1_m, is1_n, 
```

Comment:
It might be nice to refactor this to error out rather than take the slow path. The case where malloc fails is the case that will be really slow in the slow path. Erroring would require refactoring this to be a new-style inner loop function (with a return value), which could be done later.